### PR TITLE
Update tests and doctests for Julia 1.6

### LINKED ISSUE #161: CI doctest failures
https://github.com/invenia/Intervals.jl/runs/2198564468?check_suite_focus=true#step:6:1915

### PR DESCRIPTION
**Pull request by mjram0s**
Resolves #161 and #160

### CODE REVIEW

File: test/anchoredinterval.jl [hunk 1]
```diff
@@ -477,14 +531,26 @@ using Intervals: Bounded, Ending, Beginning, canonicalize, isunbounded
...
-        @test sprint(show, interval) ==
-            "AnchoredInterval{-10,$Int,Open,Closed}(10)"
+
+        if VERSION >= v"1.6.0"
+            @test sprint(show, interval) ==
+                "AnchoredInterval{-10, $Int, Open, Closed}(10)"
+        else
+            @test sprint(show, interval) ==
+                "AnchoredInterval{-10,$Int,Open,Closed}(10)"
+        end
```
Comment (omus) [inline test/anchoredinterval.jl:541]:
```suggestion
        shown = if VERSION >= v"1.6.0"
            "AnchoredInterval{-10, $Int, Open, Closed}(10)"
        else
            "AnchoredInterval{-10,$Int,Open,Closed}(10)"
        end
        @test sprint(show, interval) == shown
```

File: test/anchoredinterval.jl [hunk 2]
```diff
@@ -477,14 +531,26 @@ using Intervals: Bounded, Ending, Beginning, canonicalize, isunbounded
...
-        @test sprint(show, interval) ==
-            "AnchoredInterval{25,Char,Closed,Open}('a')"
+
+        if VERSION >= v"1.6.0"
+            @test sprint(show, interval) ==
+                "AnchoredInterval{25, Char, Closed, Open}('a')"
+        else
+            @test sprint(show, interval) ==
+                "AnchoredInterval{25,Char,Closed,Open}('a')"
+        end
```
Comment (omus) [inline test/anchoredinterval.jl:553]:
```suggestion

        shown = if VERSION >= v"1.6.0"
            "AnchoredInterval{25, Char, Closed, Open}('a')"
        else
            "AnchoredInterval{25,Char,Closed,Open}('a')"
        end
        @test sprint(show, interval) == shown
```

File: test/runtests.jl [hunk 3]
```diff
@@ -23,7 +23,7 @@ include("test_utils.jl")
     # Note: The output of the doctests currently requires a newer version of Julia
     # https://github.com/JuliaLang/julia/pull/34387
     # The doctests fail on x86, so only run them on 64-bit hardware
-    if VERSION >= v"1.5.0-DEV.163" && Sys.WORD_SIZE == 64
+    if v"1.6" >= VERSION >= v"1.7" && Sys.WORD_SIZE == 64
```
Comment (omus) [inline test/runtests.jl:26]: The previous statement is always false
```suggestion
    if v"1.6" >= VERSION && Sys.WORD_SIZE == 64
```

File: test/interval.jl [hunk 4]
```diff
@@ -282,21 +282,41 @@
         interval = Interval{Open, Open}(1, 2)
         @test string(interval) == "(1 .. 2)"
         @test sprint(show, interval, context=:compact=>true) == string(interval)
-        @test sprint(show, interval) == "Interval{$Int,Open,Open}(1, 2)"
+
+        if VERSION >= v"1.6.0"
+            @test sprint(show, interval) == "Interval{$Int, Open, Open}(1, 2)"
+        else
+            @test sprint(show, interval) == "Interval{$Int,Open,Open}(1, 2)"
+        end
```
Comment (omus) [inline test/interval.jl:290]:
```suggestion
        shown = if VERSION >= v"1.6.0"
             "Interval{$Int, Open, Open}(1, 2)"
        else
             "Interval{$Int,Open,Open}(1, 2)"
        end
        @test sprint(show, interval) == shown
```

File: test/interval.jl [hunk 5]
```diff
@@ -282,21 +282,41 @@
...
 
         interval = Interval{Open, Closed}('a', 'b')
         @test string(interval) == "(a .. b]"
         @test sprint(show, interval, context=:compact=>true) == string(interval)
-        @test sprint(show, interval) == "Interval{Char,Open,Closed}('a', 'b')"
+        if VERSION >= v"1.6.0"
+            @test sprint(show, interval) == "Interval{Char, Open, Closed}('a', 'b')"
+        else
+            @test sprint(show, interval) == "Interval{Char,Open,Closed}('a', 'b')"
+        end
```
Comment (omus) [inline test/interval.jl:299]:
```suggestion
        shown = if VERSION >= v"1.6.0"
            "Interval{Char, Open, Closed}('a', 'b')"
        else
            "Interval{Char,Open,Closed}('a', 'b')"
        end
        @test sprint(show, interval) == shown
```

File: test/interval.jl [hunk 6]
```diff
@@ -282,21 +282,41 @@
...
+            )
+        else
+            shown = string(
+                "Interval{Date,Closed,Open}(",
+                sprint(show, Date(2012, 1, 1)),
+                ", ",
+                sprint(show, Date(2013, 1, 1)),
+                ")",
+            )
+        end
```
Comment (omus) [inline test/interval.jl:319]:
```suggestion
        type_str = if VERSION >= v"1.6.0"
            "Interval{Date, Closed, Open}"
        else
            "Interval{Date,Closed,Open}"
        end
        shown = string(
            type_str,
            "(",
            sprint(show, Date(2012, 1, 1)),
            ", ",
            sprint(show, Date(2013, 1, 1)),
            ")",
        )
```

File: test/interval.jl [hunk 7]
```diff
@@ -305,8 +325,13 @@
...
         @test sprint(show, interval, context=:compact=>true) == string(interval)
-        @test sprint(show, interval) ==
-            "Interval{String,Closed,Closed}(\"a\", \"b\")"
+        if VERSION >= v"1.6.0"
+            @test sprint(show, interval) ==
+                "Interval{String, Closed, Closed}(\"a\", \"b\")"
+        else
+            @test sprint(show, interval) ==
+                "Interval{String,Closed,Closed}(\"a\", \"b\")"
+        end
```
Comment (omus) [inline test/interval.jl:334]:
```suggestion
        shown = if VERSION >= v"1.6.0"
            "Interval{String, Closed, Closed}(\"a\", \"b\")"
        else
            "Interval{String,Closed,Closed}(\"a\", \"b\")"
        end
        @test sprint(show, interval) == shown
```

**Review by omus — CHANGES_REQUESTED** [review]

File: test/runtests.jl (hunk 3 above)
Comment (mjram0s) [reply test/runtests.jl:26]: I meant to do `v"1.6" <= VERSION < v"1.7"` 🤦‍♀️

**Review by mattBrzezinski — APPROVED** [review]

**Review by omus — APPROVED** [review]

**Pull request merged**